assets/src/components/v2/disruption_diagram/disruption_diagram.tsx

```diff
@@ -743,108 +606,123 @@ const DisruptionDiagram: ComponentType<DisruptionDiagramData> = (props) => {
+  // the offset (essentially the basic height of the diagram)
+  // Otherwise, if there is vertical room, the calculated margin is subtracted from that offset
+  const viewBoxOffset = topAndBottomMargins < 0 ? offset : offset - topAndBottomMargins
 
   return (
     <svg
-      height="100%"
-      viewBox={`0 0 904 ${svgHeight}`}
-      id="whole-svg"
+      // viewBoxOffset will always be > 0 by the time it's visible, but the console will
+      // still log an error if it's a negative number when it's not-yet-visible
+      viewBox={`0 ${viewBoxOffset > 0 ? -viewBoxOffset : 0} 904 ${svgHeight}`}
+      transform={`translate(${translateX})`}
       visibility={isDone ? "visible" : "hidden"}
     >
-      <g
-        id="line-map"
-        transform={`translate(${SLOT_WIDTH / 2}, ${
-          lineMapHeight - EMPHASIS_PADDING_TOP * scaleFactor - 16
-        }) scale(${scaleFactor})`}
-      >
-        <EffectBackgroundComponent
-          effectRegionSlotIndexRange={
-            effect === "station_closure"
-              ? props.closed_station_slot_indices
-              : props.effect_region_slot_index_range
-          }
-          effect={effect}
-          spaceBetween={spaceBetween}
-        />
-        <FirstSlotComponent
-          slot={beginning}
-          line={line}
-          isCurrentStop={current_station_slot_index === 0}
-          spaceBetween={spaceBetween}
-          isAffected={
-            effect === "station_closure"
-              ? props.closed_station_slot_indices.includes(0)
-              : props.effect_region_slot_index_range.includes(0)
-          }
-          effect={effect}
-        />
-        {middleSlots}
-        <LastSlotComponent
-          slot={end as EndSlot}
-          x={x}
-          line={line}
-          isCurrentStop={current_station_slot_index === slots.length - 1}
-          isAffected={
-            effect === "station_closure"
-              ? props.closed_station_slot_indices.includes(slots.length - 1)
-              : props.effect_region_slot_index_range.includes(slots.length - 1)
-          }
-          effect={effect}
-        />
-      </g>
-      {hasEmphasis && (
-        <g
-          id="alert-emphasis"
-          transform={`translate(${SLOT_WIDTH / 2}, ${
-            lineMapHeight + EMPHASIS_HEIGHT - EMPHASIS_PADDING_TOP * scaleFactor
-          })`}
-        >
-          <AlertEmphasisComponent
-            effectRegionSlotIndexRange={props.effect_region_slot_index_range}
+      <g transform={`translate(${L * scaleFactor} 0)`}>
+        <g id="line-map" transform={`scale(${scaleFactor})`}>
+          <EffectBackgroundComponent
+            effectRegionSlotIndexRange={
+              effect === "station_closure"
+                ? props.closed_station_slot_indices
+                : props.effect_region_slot_index_range
+            }
+            effect={effect}
+            spaceBetween={spaceBetween}
+          />
```

Comment:
Would it be any easier to simply not render this component if the effect is station_closure? Right now there's some conditional stuff that happens here, as well as inside the body of `EffectBackgroundComponent`, which ultimately results in the component rendering nothing for a station_closure alert. I think that conditional logic could be pulled up here into one place, simplifying the component definition as well as its invocation since it only needs to be concerned with alert effects that actually result in a background being drawn.
```suggestion
          {effect !== "station_closure" && (
            <EffectBackgroundComponent
              effectRegionSlotIndexRange={props.effect_region_slot_index_range}
              effect={effect}
              spaceBetween={spaceBetween}
            />
          )}
```

(You might need to give TypeScript a hint that the props can only be for a continuous alert type: `{(props as ContinuousDisruptionDiagram).effect_region_slot_index_range}`)